### DISCUSSION TITLE
Removing titleNode and other nodes from ReactDialog

### DISCUSSION THREAD

**Discussion by samael-sam** [eclipse-theia/theia#10502] [General]
Hi,
I'm trying to remove `dialogTitle` and other blocks from my `CustomShell` which extends `ReactDialog`. I need help with implementing this functionality as I need to add `Menubar` in the the `dialogTitle` to the `CustomShell`.

The following is a sample code that I have tried which doesn't work. Kindly assist regarding this.

```
     export abstract class CustomShell<CustomShell> extends ReactDialog<string> {
     protected menuItem: MenuItem[];
    protected contentNode: HTMLDivElement;
    constructor(
        props: DialogProps
    ) {
        super(props);
        this.closeCrossNode.remove();
        this.controlPanel.remove();
        this.titleNode.remove();
        this.errorMessageNode.remove();

        // Remove all theia divs except this.contentNode.

        this.contentNode.classList.add('customShell');
        this.update();
    }

    get value() {
        return 'hello';
    }


    protected render() {
        return (
            <div>
                <div className='dialogTitle'>
                    <Menubar modal={this.menuItem}/> 
                    <div className='title-action-node'>
                        <i id='custom-close' className='fa fa-minus' />
                        <i id='custom-minimize' className='fa fa-times' onClick={() => { this.close() }} />
                    </div>
                </div>
                <div className='dialogContent'>
                    Hello World
                </div>
            </div>
        )
    }
}
```

Thanks

**Comment by msujew** [comment]
Hi @samael-sam, do you mind giving a bit more info on what doesn't work? I see a potential issue in there, as you define `protected contentNode: HTMLDivElement;` a second time in your component (it is already defined inside `AbstractDialog<T>`). This can lead to undefined errors and other unexpected behavior.

**Reply by samael-sam** [reply]
Hi @msujew ,
Thank you for your response. So, I am aiming to keep just the `contentNode` and remove `closeCrossNode`, `controlPanel`, `titleNode`, `errorMessageNode`, I tried removing nodes using `this.titleNode.remove()`, but this doesn't work. It's still is present in the ReactDialog. 

In short I am aiming to achieve this (Just the dialogContent): 
![image](https://user-images.githubusercontent.com/94870871/144725257-8f576a47-3cac-4fc4-876b-ab119c68800a.png)

But, with this method am getting this (The code doesn't have any effect in removing all the other nodes):
![image](https://user-images.githubusercontent.com/94870871/144725220-3c59b164-447a-41e8-b4bf-61a12a797506.png)

Also regarding `contentNode: HTMLDivElement`, I am trying to change some of its properties, since it's readonly am unable to change it value and so I thought of overriding it using the same variable in the class. Can you provide any documentation links to why it is harmful ? It would be very much helpful as I believed this was one of the way to override a variable and it would help me correct. 

Thanks
